src/controller/CommissionerDiscoveryController.cpp

```diff
@@ -0,0 +1,162 @@
+/*
+ *
+ *    Copyright (c) 2020-2021 Project CHIP Authors
+ *    Copyright (c) 2013-2017 Nest Labs, Inc.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      Declaration of Commissioner Discovery Controller,
+ *      a common class that manages state and callbacks
+ *      for handling the Commissioner Discovery
+ *      and User Directed Commissioning workflow
+ *
+ */
+#include <iostream>
```

Comment:
Is this for embedded targets? iostream tends to bring in locales, which may cause bloat. Guessing this may be the cause of the mbed running out of RAM maybe?